octodns/provider/constellix.py

```diff
@@ -82,95 +128,226 @@ def _request(self, method, path, params=None, data=None):
         time.sleep(self.ratelimit_delay)
         return resp
 
-    @property
     def domains(self):
-        if self._domains is None:
+        """
+        Retrieves a list of domains associated with the api-key's account
+        from the Constellix DNS REST API service.
+        :return:
+            Dict: Holds the domain names and their associated domain IDs.
+        """
+        if self._domains is None:   # pragma: no coverage
             zones = []
 
-            resp = self._request('GET', '/').json()
+            resp = self._request('GET', '/domains').json()
             zones += resp
 
             self._domains = {'{}.'.format(z['name']): z['id'] for z in zones}
 
         return self._domains
 
     def domain(self, name):
-        path = '/{}'.format(self.domains.get(name))
+        """
+        Retrieves the domain with the specified name from the the
+        Constellix DNS REST API service
+        :param name: string - The name of the domain to retrieve
+        :return:
+            Any: JSON representation of the domain
+        """
+        domain_name = self.domains().get(name)
+        path = '/domains/{}'.format(domain_name)
         return self._request('GET', path).json()
 
     def domain_create(self, name):
-        self._request('POST', '/', data={'names': [name]})
-
-    def _absolutize_value(self, value, zone_name):
-        if value == '':
-            value = zone_name
+        """
+        Request the creation of a domain with the specified name
+        :param name: string - The name of the domain to create
+        :return:
+            Response: The response from the API service
+        """
+        return self._request(
+            'POST',
+            '/domains', data={'names': [name]}
+        ).json()
+
+    @staticmethod
+    def _absolutize_value(value, zone_name):
+        """
+        Converts the fully-qualified domain name (FQDN) into its absolute
+        form by concatenating the sub-domain and the domain. Note that
+        the `zone_name` is required to have a trailing dot (otherwise the
+        zone object cannot be created).
+        :param value: string - The sub-domain name
+        :param zone_name: string - The domain name (required dot at end)
+        :return:
+            string: The absolute version of the FQDN
+        """
+        if value == '' or value == '.':
+            fqdn = zone_name
         elif not value.endswith('.'):
-            value = '{}.{}'.format(value, zone_name)
+            fqdn = '{}.{}'.format(value, zone_name)
+        else:
```

Comment:
This is unnecessary and causes problems when the value is to something not within the same zone

```
* constellix (ConstellixProvider)
*   Update
*     <CnameRecord CNAME 3600, calendar.yzguy.io., ghs.googlehosted.com.yzguy.io.> ->
*     <CnameRecord CNAME 3600, calendar.yzguy.io., ghs.googlehosted.com.> (config)
```